testutils/blockdata.go

```diff
@@ -0,0 +1,123 @@
+package testutils
+
+import (
+	"bytes"
+	"github.com/celestiaorg/celestia-core/types"
+	"math"
+	"math/rand"
+	"sort"
+	"time"
+
+	tmbytes "github.com/celestiaorg/celestia-core/libs/bytes"
+	"github.com/celestiaorg/celestia-core/types/consts"
+)
+
+// TODO eventually move this to a testutils package
+
+// GenerateRandomBlockData returns randomly generated block data for testing purposes
+func GenerateRandomBlockData(txCount, isrCount, evdCount, msgCount, maxSize int) types.Data {
+	var out types.Data
+	out.Txs = GenerateRandomlySizedContiguousShares(txCount, maxSize)
+	out.IntermediateStateRoots = generateRandomISR(isrCount)
+	out.Evidence = generateIdenticalEvidence(evdCount)
+	out.Messages = GenerateRandomlySizedMessages(msgCount, maxSize)
+	return out
+}
+
+func GenerateRandomlySizedContiguousShares(count, max int) types.Txs {
+	txs := make(types.Txs, count)
+	for i := 0; i < count; i++ {
+		size := rand.Intn(max)
+		if size == 0 {
+			size = 1
+		}
```

Comment:
```suggestion
		// ensure that no transactions are 0 bytes, as no valid transaction has only 0 bytes
		if size == 0 {
			size = 1
		}
```

Reply:
alternatively, we could just use a different mechanism to prevent 0. iirc, this was just done because it was the fastest and required the least amount of thinking